packages/@react-stately/table/src/useTableColumnResizeState.ts

```diff
@@ -1,226 +1,148 @@
-
-    setColumnWidthsForRef(widths);
-
-    /*
-     when getting recalculated columns above, the column being resized is not considered "recalculated"
-     so we need to add it to the list of affected columns
-    */
-    let allAffectedColumns = ([[column.key, boundedWidth], ...recalculatedColumnWidths] as [Key, number][]).map(([key, width]) => ({key, width}));
-    return allAffectedColumns;
-  }
-
-  // This function is regenerated whenever columnWidthsRef.current changes in order to get the new correct ref value.
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  let getColumnWidth = useCallback((key: Key): number => columnWidthsRef.current.get(key) ?? 0, [columnWidthsRef.current]);
-
-  let getColumnMinWidth = useCallback((key: Key) => {
-    const columnIndex = columns.findIndex(col => col.key === key);
-    if (columnIndex === -1) {
-      return;
-    }
-    return getMinWidth(columns[columnIndex].props.minWidth, tableWidth.current);
-  }, [columns]);
-
-  let getColumnMaxWidth = useCallback((key: Key) => {
-    const columnIndex = columns.findIndex(col => col.key === key);
-    if (columnIndex === -1) {
-      return;
-    }
-    return getMaxWidth(columns[columnIndex].props.maxWidth, tableWidth.current);
-  }, [columns]);
-
-  return {
-    columnWidths: columnWidthsRef,
-    setTableWidth,
+export function useTableColumnResizeState<T>(props: TableColumnResizeStateProps<T>, state): TableColumnResizeState {
+  let {
+    getDefaultWidth,
+    getDefaultMinWidth,
+    onColumnResizeStart: propsOnColumnResizeStart,
+    onColumnResizeEnd: propsOnColumnResizeEnd
+  } = props;
+
+  let columnLayout = useMemo(
+    () => new TableColumnLayout({
+      getDefaultWidth,
+      getDefaultMinWidth
+    }),
+    [getDefaultWidth, getDefaultMinWidth]
+  );
+
+  let tableWidth = props.tableWidth ?? 0;
+  let [controlledColumns, uncontrolledColumns]: [Map<Key, GridNode<unknown>>, Map<Key, GridNode<unknown>>] = useMemo(() =>
+    splitColumnsIntoControlledAndUncontrolled(state.collection.columns)
+  , [state.collection.columns]);
+
+  // uncontrolled column widths
+  let [uncontrolledWidths, setUncontrolledWidths] = useState<Map<Key, ColumnSize>>(() =>
+    getInitialUncontrolledWidths(uncontrolledColumns, columnLayout)
+  );
+  // combine columns back into one map that maintains same order as the columns
+  let cWidths = useMemo(() =>
+    recombineColumns(state.collection.columns, uncontrolledWidths, uncontrolledColumns, controlledColumns)
+  , [state.collection.columns, uncontrolledWidths, uncontrolledColumns, controlledColumns]);
+
+  let onColumnResizeStart = useCallback((key: Key) => {
+    columnLayout.setResizingColumn(key);
+    propsOnColumnResizeStart?.(key);
+  }, [columnLayout, propsOnColumnResizeStart]);
+
+  // TODO: move props.on* all into this file and layout, or move them all out to the aria handler..., stately would be preferable
+  let onColumnResize = useCallback((key: Key, width: number): Map<Key, ColumnSize> => {
```

Comment:
Is this TODO still valid?

Reply:
I would like it to still be valid, I can't figure out a way to get the events into TableLayout though. I can't create a new TableLayout object every time one of the onResize* functions changes, people write stuff like this all the time
```
<TableView onResize={(widths) => myHandler(widths)} ...
// or
let onResize = () => {...}
<TableView onResize={onResize}
```
both of which would invalidate the TableLayout every render if I added them to the memo

As a result, I can't get them into the state.

I thought about using a ref to store the incoming props, but then i'd have to write to the ref during render, which is a no-no